### ISSUE TITLE
SA1012 incorrectly reported on opening brace in List Pattern matching

### ISSUE THREAD

**Issue by LWehmschulteAtRosenxt** [DotNetAnalyzers/StyleCopAnalyzers#3858]
```csharp
 public void T()
{
    var a = Enumerable.Range(0, 4).Select(x => new A() { B = x }).ToList();
               
// ⚠️ SA1012 An opening brace within a C# element is not spaced correctly.
//            ↓
    if (a is [{ B: var b }, ..])
    {
        Console.WriteLine(b);
    }
}

private class A()
{
    #region Properties

    public int B { get; set; }

    #endregion
} 
```

**Comment by bjornhellander** [comment]
I can't reproduce this with the latest release (1.2.0-beta.556). Are you using 1.1.118?

1.1.118 is from 2019 and does not handle modern c# features well. There hasn't been an official release since then due to a combination of project policy and not enough people helping out, so if you want to use modern c#, the recommendation is to use the latest beta.

**Comment by LWehmschulteAtRosenxt** [comment]
> I can't reproduce this with the latest release (1.2.0-beta.556). Are you using 1.1.118?
> 
> 1.1.118 is from 2019 and does not handle modern c# features well. There hasn't been an official release since then due to a combination of project policy and not enough people helping out, so if you want to use modern c#, the recommendation is to use the latest beta.

My bad 😬 I've updated the Nuget package reference through our Sdk.props file but the changes didn't come through yet. After a Visual Studio restart everything appears to work fine
